pkg/auth/oauth2/auth.go

```diff
@@ -297,6 +304,43 @@ func (a *OAuth2Authenticator) LoginFunc(w http.ResponseWriter, r *http.Request)
 	http.Redirect(w, r, a.oauth2Config().AuthCodeURL(state), http.StatusSeeOther)
 }
 
+func (a *OAuth2Authenticator) ReviewToken(r *http.Request) error {
+	token, err := sessions.GetSessionTokenFromCookie(r)
+	if err != nil {
+		return err
+	}
+
+	tokenReview := &authv1.TokenReview{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "authentication.k8s.io/v1",
+			Kind:       "TokenReview",
+		},
+		Spec: authv1.TokenReviewSpec{
+			Token: token,
+		},
+	}
+
+	completedTokenReview, err := a.
+		internalK8sClientset.
+		AuthenticationV1().
+		TokenReviews().
+		Create(r.Context(), tokenReview, metav1.CreateOptions{})
+
+	if err != nil {
+		return fmt.Errorf("failed to create TokenReview, %v", err)
+	}
+
+	// Check if the token is authenticated
+	if !completedTokenReview.Status.Authenticated {
+		if completedTokenReview.Status.Error != "" {
+			return errors.New(completedTokenReview.Status.Error)
+		}
+		return errors.New("unknown error")
```

Comment:
lets add more descriptive error
```suggestion
		return errors.New("failed to authenticate the token, with unknown error")
```